libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_Scripting/AP_Scripting.h>
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+
+#include "lua_bindings.h"
+
+// ensure that we have a set of stack sizes, and enforce constraints around it
+// except for the minimum size, these are allowed to be defined by the build system
+#undef SCRIPTING_STACK_MIN_SIZE
+#define SCRIPTING_STACK_MIN_SIZE 2048
+
+#if !defined(SCRIPTING_STACK_SIZE)
+  #define SCRIPTING_STACK_SIZE 8192
+#endif // !defined(SCRIPTING_STACK_SIZE)
+
+#if !defined(SCRIPTING_STACK_MAX_SIZE)
+  #define SCRIPTING_STACK_MAX_SIZE 16384
+#endif // !defined(SCRIPTING_STACK_MAX_SIZE)
+
+static_assert(SCRIPTING_STACK_SIZE >= SCRIPTING_STACK_MIN_SIZE, "Scripting requires a larger minimum stack size");
+static_assert(SCRIPTING_STACK_SIZE <= SCRIPTING_STACK_MAX_SIZE, "Scripting requires a smaller stack size");
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_Scripting::var_info[] = {
+    // @Param: ENABLE
```

Comment:
Will this be a bitmask in the future? I mean one bit per scripting language ?

Reply:
Not within any near term future. A lot of flash space goes into the runtimes, as well as memory when running. We are of course free to change that in the future, but it doesn't need to be committed to now. I should note that at this point because it's opt in (and not meant to be flown yet) I'm not committing to that param table not being rearranged.